packages/gensx-anthropic/src/stream.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import {
+  Message,
+  MessageCreateParamsNonStreaming,
+  RawMessageStreamEvent,
+  ToolUseBlock,
+} from "@anthropic-ai/sdk/resources/messages";
+import { Stream } from "@anthropic-ai/sdk/streaming";
+import { gsx } from "gensx";
+
+import { AnthropicChatCompletion } from "./anthropic.js";
+import { GSXTool } from "./tools.js";
+import { toolExecutorImpl } from "./tools.js";
+
+type StreamCompletionProps = Omit<
+  MessageCreateParamsNonStreaming,
+  "stream" | "tools"
+> & {
+  stream: true;
+  tools?: GSXTool<any>[];
+};
+
+type StreamCompletionOutput = Stream<RawMessageStreamEvent>;
+
+export const streamCompletionImpl = async (
+  props: StreamCompletionProps,
+): Promise<StreamCompletionOutput> => {
+  const { stream, tools, ...rest } = props;
+
+  // If we have tools, first make a synchronous call to get tool calls
+  if (tools?.length) {
+    // Make initial completion to get tool calls
+    const completion = await gsx.execute<Message>(
+      <AnthropicChatCompletion
+        {...rest}
+        tools={tools.map((t) => t.definition)}
+        stream={false}
+      />,
+    );
+
+    // If no tool calls, proceed with streaming the original response
+    if (completion.stop_reason !== "tool_use") {
+      return gsx.execute<Stream<RawMessageStreamEvent>>(
+        <AnthropicChatCompletion {...rest} stream={true} />,
+      );
```

Comment:
You can use Component.run now (might need to rebase).

```suggestion
      return AnthropicChatCompletion.run({
        ...rest,
        stream: true
      });
```